### DISCUSSION TITLE
Record TTL input for stimulation timing

### DISCUSSION THREAD

**Discussion by TomVCoopmans** [AllenInstitute/MIES#2135] [Q&A] [answered]
Hi,
I would like to record the TTL input of the DAC (ITC 1600). Our imaging program (ScanImage) can sent TTLs when stimulation starts and I would like to record those with a TTL input on the DAC, to temporally correlate the stimulus to the Ephys readings. Is this possible? Currently, MIES only supports (as far as we know) TTL out, to externally trigger devices.

My workaround is now to have the TTL as a AD input and record that channel. However, this leads to lower acquisition rate of the DAC and an "analog" signal that is harder for me to correlate in my analysis. Is it possible to have MIES record a TTL with data acquisition?

Thank you for the help. Please let me know if you need more information.

Tom Coopmans
Goriounova Lab, Vrije Universiteit Amsterdam

**Comment by t-b** [comment, marked as answer]
Thanks for contacting us.

Yes there is no support for "TTL in" for any hardware. I'm also not sure that adding "TTL in" support would have the benefits your are envisioning. The data we sent/receive from the ITC hardware is always 16bit integer and we convert that to float/double as requested. So also the "TTL in" data would be 16bit integer initially.

If you want to acquire with a fixed acquisition rate (DA_Ephys panel->Settings->Sampling Frequency) you can also do that no matter what the active channels are. I also guess that enabling one "TTL in" channel would influence the minimum sampling interval like enabling an additional AD channel.

For the analysis of the digital signal on the AD channel it should be easy to find the edges with FindLevels.

**Reply by TomVCoopmans** [reply]
Thank you for the quick response. I will use the AD channel input and use FindLevels.
